chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/MergeDeveloperJob.java

```diff
@@ -0,0 +1,330 @@
+package gov.healthit.chpl.scheduler.job;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.AddressException;
+import javax.transaction.Transactional;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.quartz.Job;
+import org.quartz.JobDataMap;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.context.support.SpringBeanAutowiringSupport;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import gov.healthit.chpl.auth.user.JWTAuthenticatedUser;
+import gov.healthit.chpl.dao.DeveloperDAO;
+import gov.healthit.chpl.domain.CertifiedProductSearchDetails;
+import gov.healthit.chpl.domain.activity.ActivityConcept;
+import gov.healthit.chpl.dto.CertifiedProductDetailsDTO;
+import gov.healthit.chpl.dto.DeveloperDTO;
+import gov.healthit.chpl.dto.ProductDTO;
+import gov.healthit.chpl.dto.ProductOwnerDTO;
+import gov.healthit.chpl.dto.auth.UserDTO;
+import gov.healthit.chpl.exception.EntityCreationException;
+import gov.healthit.chpl.exception.EntityRetrievalException;
+import gov.healthit.chpl.manager.ActivityManager;
+import gov.healthit.chpl.manager.CertifiedProductDetailsManager;
+import gov.healthit.chpl.manager.CertifiedProductManager;
+import gov.healthit.chpl.manager.DeveloperManager;
+import gov.healthit.chpl.manager.ProductManager;
+import gov.healthit.chpl.scheduler.SchedulerCertifiedProductSearchDetailsAsync;
+import gov.healthit.chpl.service.DirectReviewUpdateEmailService;
+import gov.healthit.chpl.util.EmailBuilder;
+import net.sf.ehcache.CacheManager;
+
+public class MergeDeveloperJob implements Job {
+    public static final String JOB_NAME = "mergeDeveloperJob";
+    public static final String OLD_DEVELOPERS_KEY = "oldDevelopers";
+    public static final String NEW_DEVELOPER_KEY = "newDeveloper";
+    public static final String PRODUCT_IDS_TO_MOVE_KEY = "productIdsToMove";
+    public static final String USER_KEY = "user";
+    private static final Logger LOGGER = LogManager.getLogger("mergeDeveloperJobLogger");
```

Comment:
I think Andrew showed me that you can use `@Log4j2(topic="mergeDeveloperJobLogger")` for this.